SmartDeviceLink/private/SDLMenuManager.m

```diff
@@ -551,8 +551,17 @@ - (BOOL)sdl_isDynamicMenuUpdateActive:(SDLDynamicMenuUpdatesMode)dynamicMenuUpda
     return [mutableArtworks allObjects];
 }
 
-- (BOOL)sdl_artworkNeedsUpload:(SDLArtwork *)artwork {
-    return (artwork != nil && ![self.fileManager hasUploadedFile:artwork] && !artwork.isStaticIcon);
+- (BOOL)sdl_areAllCellArtworksUploaded:(NSArray<SDLMenuCell *> *)cells {
+    for (SDLMenuCell *cell in cells) {
+        SDLArtwork *artwork = cell.icon;
+        if (artwork != nil && !artwork.isStaticIcon && self.fileManager != nil && ![self.fileManager hasUploadedFile:(artwork)]) {
```

Comment:
```suggestion
        if (artwork != nil && !artwork.isStaticIcon && ![self.fileManager hasUploadedFile:artwork]) {
```

In addition, instead of checking `artwork != nil` here, the `fileManager hasUploadedFile:` method should return `NO` if the passed artwork is `nil`. Additionally, the implementation of `hasUploadedFile` uses `true` and `false` instead of `YES` and `NO`. Could you please fix that?